src/goto-symex/symex_main.cpp

```diff
@@ -332,12 +332,75 @@ void goto_symext::symex_step(reachability_treet &art)
     if (is_symbol2t(call.function))
     {
       const irep_idt &id = to_symbol2t(call.function).thename;
-      if (has_prefix(id.as_string(), "c:@F@__ESBMC"))
+      std::string id_str = id.as_string();
+
+      // Handle __ESBMC intrinsic functions with both C and Python prefixes.
+      // Python frontend generates function names like "py:1.py@F@__ESBMC_get_object_size"
+      // which need to be converted to "c:@F@__ESBMC_get_object_size" format.
+      if (has_prefix(id_str, "c:@F@__ESBMC"))
```

Comment:
This solution might slow down our symex.

I suspect this might be the cause: https://github.com/esbmc/esbmc/blob/master/src/python-frontend/preprocessor.py#L903.

Can you try to intercept this call at https://github.com/esbmc/esbmc/blob/master/src/python-frontend/function_call_builder.cpp#L56?

You can check some examples:

strings:
https://github.com/esbmc/esbmc/blob/master/src/python-frontend/function_call_builder.cpp#L228

lists:
https://github.com/esbmc/esbmc/blob/master/src/python-frontend/function_call_builder.cpp#L249

dictionaries:
https://github.com/esbmc/esbmc/blob/master/src/python-frontend/function_call_builder.cpp#L278